readme.md

```diff
@@ -133,10 +134,97 @@ Here is further explanation on what each profile starts.
 |   Schema registry    |   x   |   x   |   x   |            |               |
 |      Opensearch      |   x   |   x   |       |     x      |               |
 | Opensearch dashboard |   x   |       |       |     x      |               |
+| Opensearch configurer|   x   |   x   |   x   |     x      |       x       |
 |      Prometheus      |   x   |       |       |            |       x       |
 |       Grafana        |   x   |       |       |            |       x       |
+
+## Keycloak
+
+Keycloak is used as a local identity provider, to be able to mimic a production security model with service to service authentication.
+
+### Useful urls:
+
+- OpenID Endpoint Configuration: http://localhost:8080/realms/local-development/.well-known/openid-configuration
+- Token Endpoint http://localhost:8080/realms/local-development/protocol/openid-connect/token 
+
+### Default clients:
+
+A set of default clients have been defined which covers most common usecases.
+
+All roles are mapped to the `roles` claim in the JWT. This configuration is defined in [local-development.json](./config/keycloak/local-development.json) and is applied to keycloak using the `keycloak-setup` service.
+To modify the configuration either go to the [admin console](http://localhost:8080/admin) (Username: `admin` Password: `admin`) or edit the `local-development.json` following this [guide](./config/keycloak/setup.md)
+
+- Default access
+     * Description: Read and write access to all data Kafka, OpenSearch and Schema registry
+     * client_id: `default-access`
+     * client_secret: `default-access-secret`
+     * default_scopes: [ ] 
+     * optional_scopes:
+          - `kafka`
+               * Roles:
+                    - `Kafka_*_all`
+          - `opensearch`
+               * Roles:
+                    - `opensearch_default_access`
+          - `schema-registry`
+               * Roles:
+                    - `sr-producer`
+- Default write
+     * Description: Write access to all data in Kafka, OpenSearch and Schema registry
+     * client_id: `default-write`
+     * client_secret: `default-write-secret`
+     * default_scopes: [ ] 
+     * optional_scopes:
+          - `kafka`
+               * Roles:
+                    - `Kafka_*_write`
+          - `opensearch`
+               * Roles:
+                    - `opensearch_default_write`
+          - `schema-registry`
+               * Roles:
+                    - `sr-producer`
+- Default read
+     * Description: Read access to all data in Kafka, OpenSearch and Schema registry
+     * client_id: `default-read`
+     * client_secret: `default-read-secret`
+     * default_scopes: [ ] 
+     * optional_scopes:
+          - `kafka`
+               * Roles:
+                    - `Kafka_*_read`
+          - `opensearch`
+               * Roles:
+                    - `opensearch_default_read`
+- Users
+     * Description: User login via browser (See [Users](#users) for user details)
```

Comment:
is this used ?

Reply:
Yes. It can be used to login to login to the opensearch dashboard.

Reply:
Should we make it a bit more explcit?

Reply:
```suggestion
     * Description: User login via browser such as OpenSearch Dashboard (See [Users](#users) for user details)
```

Reply:
Good idea